common/build/build-common/tsconfig.test.node16.json

```diff
@@ -2,6 +2,7 @@
 	"extends": ["./tsconfig.base.json", "./tsconfig.node16.json"],
 	"compilerOptions": {
 		"composite": false,
+		"customConditions": ["allow-ff-test-exports"],
```

Comment:
by only putting this in the test config, we get good safety against including test-only exports in production code: ts won't compile.

Reply:
I wasn't aware of this option - this is cool! I think we probably need to document this pattern somewhere, though, if we want to adopt it. The package README probably?

Reply:
I also wonder if we want to do something similar for our `/internal` exports as an additional layer of protection against external consumers using our internal exports. We could communicate the pattern workaround to our internal partners in edge cases where they need to temporarily use our internal exports. @tylerbutler @DLehenbauer @jason-ha interested in your thoughts on this.

Reply:
It's not a great experience for users of our packages to do this sort of thing on code we expect them to reference (which I'm assuming is the case for /internal, albeit transitively, since our production codepath will contain such imports). They'd need to add in a bunch of conditions at bundle / runtime a la what we have here.

Reply:
We cannot nicely encapsulate the boilerplate. If we use a pattern like this for `/internal` exports, we will also break customers using our packages (because our packages import from `/internal`, and those imports will not work without the customer turning on the condition)